### PR TITLE
Support throttling /requestshow

### PR DESCRIPTION
**Pull request by mia-pi-git**
This was requested by the staff of Arts & Culture, since apparently people often /requestshow too much.

### CODE REVIEW

**Comment by mia-pi-git** [comment]
Bad forcepush.

File: server/chat-commands/info.ts [hunk 1]
```diff
@@ -2440,6 +2440,12 @@ export const commands: ChatCommands = {
 		if (!/^https?:\/\//.test(link)) link = `https://${link}`;
 		link = encodeURI(link);
 		if (!room.pendingApprovals) room.pendingApprovals = new Map();
+		if (!room.settings.showThrottle) room.settings.showThrottle = 0;
+		if (!room.lastShowRequest) room.lastShowRequest = {};
+		if (room.lastShowRequest[user.id] && Date.now() - room.lastShowRequest[user.id] <= room.settings.showThrottle) {
+			return this.errorReply(`Your request was not submitted because you've been sending them too quickly.`);
```
Comment (Zarel) [inline server/chat-commands/info.ts:2446]: "You" is not really accurate. Is it supposed to be a global timer and not a per-user timer?

**Comment by Zarel** [comment]
I'd like to speak to the staff directly.

File: server/chat-commands/info.ts (hunk 1 above)
Comment (mia-pi-git) [reply server/chat-commands/info.ts:2446]: It's supposed to be per-user, yes.

Comment (Zarel) [reply server/chat-commands/info.ts:2446]: Your current implementation is a global timer, in case that was unclear.

**Comment by mia-pi-git** [comment]
Discussed, not worth it. Closing.

**Comment by Zarel** [comment]
Specifically: Most room staff mentioned they don't really need it. Some pointed out that there are a few situations where a user has multiple related images and should be able to post them all at once.

We also noted:

- `/mute` can already prevent people from using `/requestshow`, if they're spamming it
- You can only `/requestshow` again after your previous `/requestshow` is accepted/denied, so there's no risk of a single user creating a huge backlog or otherwise spamming multiple requests per second

**Pull request closed without merging**